libsolidity/analysis/TypeChecker.cpp

```diff
@@ -299,33 +299,50 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 		if (!_function.isConstructor() && !_function.isFallback() && !_function.isPartOfExternalInterface())
 			m_errorReporter.typeError(_function.location(), "Internal functions cannot be payable.");
 	}
-	for (ASTPointer<VariableDeclaration> const& var: _function.parameters() + _function.returnParameters())
-	{
-		if (type(*var)->category() == Type::Category::Mapping)
+	auto checkArgumentAndReturnParameter = [&](VariableDeclaration const& var) {
```

Comment:
Can you split this refactoring and the `isInterface` out? I think that can be merged separately.

Reply:
I split it into three separate commits for now - do you think separate PRs are necessary?

Reply:
Merged in #5817.